src/EFCore.PG.NTS/Query/ExpressionTranslators/Internal/NpgsqlNetTopologySuiteMethodCallTranslatorPlugin.cs

```diff
@@ -42,8 +42,10 @@ public virtual Expression Translate(MethodCallExpression e, IDiagnosticsLogger<D
                 return new SqlFunctionExpression("ST_AsBinary",      typeof(byte[]),   new[] { e.Object });
             case "AsText":
                 return new SqlFunctionExpression("ST_AsText",        typeof(string),   new[] { e.Object });
-            case "Buffer":
+            case "Buffer" when e.Arguments.Count == 1:
                 return new SqlFunctionExpression("ST_Buffer",        typeof(Geometry), new[] { e.Object, e.Arguments[0] });
+            case "Buffer" when e.Arguments.Count == 2:
+                return new SqlFunctionExpression("ST_Buffer",        typeof(Geometry), new[] { e.Object, e.Arguments[0], e.Arguments[1] });
```

Comment:
Can you keep a single case with something like this?

```c#
new[] { e.Object }.Concat(e.Arguments));
```

We do it elsewhere, but not a big deal if you can't (or have a reason to avoid it) here.

Reply:
Should be fixed